libs/@blockprotocol/graph/src/stdlib/bound.ts

```diff
@@ -0,0 +1,83 @@
+import {
```

Comment:
@TimDiekmann making a comment so we can thread replies

> I'd prefer to have at least some test cases in place as the logic is not easy.

I'd prefer that too, I wanted to discuss the decision to defer tests, but had to jump off yesterday. I've updated the PR description with my thoughts about it:

> - I wanted to write a proper suite of tests to check this, but made the uncomfortable compromise at the moment to defer that until later on, to unblock some further work. We intend to write a test suite for the `Subgraph` library, and I'd encourage us to write tests for temporal operations at the same time. 
 > - It's worth noting that these temporal operations were ported from some logic we had implemented [for the HASH graph query layer ](https://github.com/hashintel/hash/blob/795bcf1e4f73c2c186da576b314d0a3ffc5bfa8d/apps/hash-graph/lib/interval-ops/src/interval.rs#L11), which are heavily tested, so confidence is fairly high that these will be correct.